go/common/node/node.go

```diff
@@ -145,19 +145,22 @@ type nodeV2 struct { // nolint: maligned
 
 // ToV3 returns the V3 representation of the V2 node descriptor.
 func (nv2 *nodeV2) ToV3() *Node {
-	return &Node{
+	nv3 := &Node{
 		Versioned:       cbor.NewVersioned(3),
 		ID:              nv2.ID,
 		EntityID:        nv2.EntityID,
 		Expiration:      nv2.Expiration,
 		P2P:             nv2.P2P,
 		Consensus:       nv2.Consensus,
-		VRF:             nv2.VRF,
```

Comment:
Not sure if zero value VRF could be a problem here. We could also blacklist zero value signing key.

Reply:
Verification will fail for small-order keys and you need to sign your node registration. So using an all-zero signing key will not get you far.